src/NHibernate/Type/CollectionType.cs

```diff
@@ -471,16 +471,39 @@ public override bool IsDirty(object old, object current, bool[] checkable, ISess
 		}
 
 		/// <summary>
-		/// Get the key value from the owning entity instance.
+		/// Get the key value from the owning entity instance. It is usually the identifier, but it might be some
+		/// other unique key, in the case of a property-ref.
 		/// </summary>
 		public object GetKeyOfOwner(object owner, ISessionImplementor session)
```

Comment:
Ceasing to return the referenced property value from this method causes bug in the many-to-many case, causing it to insert in the many-to-many table the identifier instead of the referenced property value.